java/opendcs/src/main/java/decodes/tsdb/algo/RunningAverageAlgorithm.java

```diff
@@ -37,39 +37,31 @@
 import decodes.tsdb.DbCompException;
 import decodes.tsdb.DbIoException;
 import decodes.tsdb.VarFlags;
+import org.opendcs.annotations.algorithm.Algorithm;
+import org.opendcs.annotations.algorithm.Input;
+import org.opendcs.annotations.algorithm.Output;
 
-//AW:JAVADOC
-/**
-RunningAverageAlgorithm averages single 'input' parameter to a single 'average' 
-parameter. A separate aggPeriodInterval property should be supplied.
-Example, input=Hourly Water Level, output=Daily Running Average, computed hourly,
-so each hour's output is the average of values at [t-23h ... t].
- */
-//AW:JAVADOC_END
+
+@Algorithm(description ="RunningAverageAlgorithm averages single 'input' parameter to a single 'average'\n" + 
+"parameter. A separate aggPeriodInterval property should be supplied.\n" +
+"Example, input=Hourly Water Level, output=Daily Running Average, computed hourly,\n" +
+"so each hour's output is the average of values at [t-23h ... t]." )
 public class RunningAverageAlgorithm extends decodes.tsdb.algo.AW_AlgorithmBase
 {
-//AW:INPUTS
-	public double input;	//AW:TYPECODE=i
-	String _inputNames[] = { "input" };
-//AW:INPUTS_END
+	@Input public double input;	//AW:TYPECODE=i
+	@Input String _inputNames[] = { "input" };
 
-//AW:LOCALVARS
 	double tally;
 	int count;
 	Date lastTimeSlice = null;
 
-//AW:LOCALVARS_END
 
-//AW:OUTPUTS
-	public NamedVariable average = new NamedVariable("average", 0);
-	String _outputNames[] = { "average" };
-//AW:OUTPUTS_END
+	@Output public NamedVariable average = new NamedVariable("average", 0);
+	@Output String _outputNames[] = { "average" };
 
-//AW:PROPERTIES
-	public long minSamplesNeeded = 1;
-	public boolean outputFutureData = false;
-	String _propertyNames[] = { "minSamplesNeeded", "outputFutureData" };
-//AW:PROPERTIES_END
+	@PropertySpec public long minSamplesNeeded = 1;
```

Comment:
Due to a limitation of the java annotation system, the default value can't be picked up by the annotation processor, each `@PropertySpec` needs to have the value set manually. Example below. All of the default values are a string representation of the value. Generally you can copy the value as in between some quotes.

```suggestion
	@PropertySpec(value = "1") public long minSamplesNeeded = 1;
```

Reply:
When it's a string like (public String method = "interp";), should I add something like (@PropertySpec(String = "Interp") public blah blah

Reply:
It would be the same `@PropertySpec(value = ""interp")` regardless of the type. (The system doesn't currently deal with the property type.